official/vision/data/tfrecord_lib.py

```diff
@@ -168,7 +168,8 @@ def write_tf_record_dataset(output_path, annotation_iterator,
       logging.info('On image %d', idx)
 
     total_num_annotations_skipped += num_annotations_skipped
-    writers[idx % num_shards].write(tf_example.SerializeToString())
+    if tf_example:
```

Comment:
If `tf_example_iterator` doesn't return anything, the loop simply won't run. Do we really need this `if` condition ?

Reply:
yes, `if tf_example` is not required but if in case the feature dict has image meta data but doesn't have any data for bounding box annotations still we are writing to `tfrecords`. If the image doesn't have any bounding box annotations/ skipped due to wrong annotations should we still need to write that image data to output `tfrecords`

Reply:
As I mentioned, could you help add a test case to cover this edge condition.